### PR TITLE
Fix servers transport not found

### LINKED ISSUE #7809: "servers transport not found default@internal" error when using file provider
Traefik shows following error when using file provider but continue to work properly

```
ERRO[2021-01-27T19:42:10+03:00] servers transport not found default@internal  entryPointName=web routerName=load-balancing-demo@file
servers transport not found
```

I'm using Traefik version 2.4.0 installed via brew (also tried to download a binary from the official site)
OS: MacOS BigSur 11.0.1 (20B29)

Command used to run Traefik
```
traefik --configfile demo.yaml
```

Static config demo.yaml

```yaml
entryPoints:
  web:
    address: ":80"
  traefik:
    address: ":9000"

log: {}

accessLog: {}

api:
  insecure: true
  dashboard: true

metrics:
  prometheus: {}

providers:
  file:
    filename: load-balancing.yaml
    watch: true
```

Dynamic config load-balancing.yaml

```yaml
http:
  routers:
    load-balancing-demo:
      entryPoints:
        - web
      service: load-balanced-svc
      rule: Host(`demo.local`)

  services:
    load-balanced-svc:
      loadBalancer:
        servers:
          - url: localhost:3001
```

### PR DESCRIPTION
**Pull request by jspdown**
### What does this PR do?

This PR fixes `servers transport not found default@internal`. 

### Motivation

Fixes #7809 

### More

- [x] Added/updated tests
- [ ] ~Added/updated documentation~

### CODE REVIEW

**Review by ldez — APPROVED** [review]
LGTM

**Review by rtribotte — APPROVED** [review]
LGTM 👌

**Review by jbdoumenjou — APPROVED** [review]
LGTM :+1:

File: pkg/provider/aggregator/aggregator_test.go [hunk 1]
```diff
@@ -33,8 +33,8 @@ func TestProviderAggregator_Provide(t *testing.T) {
 	}()
 
 	// Make sure the file provider is always called first, followed by the internal provider.
```
Comment (lwis) [inline pkg/provider/aggregator/aggregator_test.go:35]: Is this comment still true? Would be good to have some justification here also.

**Pull request merged**